lib/foreman_remote_execution_core/script_runner.rb

```diff
@@ -27,8 +27,8 @@ def start
 
       # pipe the output to tee while capturing the exit code in a file
       script = <<-SCRIPT
-          (#{su_prefix}#{remote_script} < /dev/null; echo $?>#{exit_code_path}) | /usr/bin/tee #{output_path}
```

Comment:
```shell
$ dpkg -l | grep csh
ii  csh                     20110502-2.2ubuntu1    amd64        Shell with C-like syntax
ii  tcsh                    6.20.00-7              amd64        TENEX C Shell, an enhanced version of Berkeley csh
```
It seems as if `csh` is just a symlink to `tcsh` on your machine. Now when I'm thinking about it `$status` in `sh` worked for me because it is probably a symlink to `bash` on my machine...

How about we won't try to support various shells and just wrap the whole script in `/bin/sh -c '#{script}'`? `/bin/sh` should be present and behave roughly the same way everywhere